### PR TITLE
Pre-signed URLs for Claim recordings

### PR DESCRIPTION
**Pull request by hedvigoscar**

### CODE REVIEW

File: src/main/java/com/hedvig/backoffice/graphql/types/Claim.java [hunk 1]
```diff
@@ -1,7 +1,10 @@
 package com.hedvig.backoffice.graphql.types;
 
+import com.amazonaws.services.s3.AmazonS3;
 import com.hedvig.backoffice.services.claims.dto.ClaimData;
 import lombok.Value;
+import lombok.val;
```
Comment (meletios) [inline src/main/java/com/hedvig/backoffice/graphql/types/Claim.java]: same here

File: src/main/java/com/hedvig/backoffice/graphql/types/Claim.java [hunk 2]
```diff
@@ -1,7 +1,10 @@
 package com.hedvig.backoffice.graphql.types;
 
+import com.amazonaws.services.s3.AmazonS3;
```
Comment (meletios) [inline src/main/java/com/hedvig/backoffice/graphql/types/Claim.java]: you can discard the imports on this file.It seems that are not needed

File: src/main/java/com/hedvig/backoffice/graphql/types/Claim.java (hunk 1 above)
Comment (hedvigoscar) [reply src/main/java/com/hedvig/backoffice/graphql/types/Claim.java]:
```suggestion
```

File: src/main/java/com/hedvig/backoffice/graphql/types/Claim.java (hunk 2 above)
Comment (hedvigoscar) [reply src/main/java/com/hedvig/backoffice/graphql/types/Claim.java]:
```suggestion
```

File: src/main/java/com/hedvig/backoffice/services/claims/ClaimsServiceImpl.java [hunk 3]
```diff
@@ -121,7 +131,18 @@ public long totalClaims(String token) {
 
   @Override
   public List<Claim> getClaimsByIds(List<UUID> ids) {
-    return client.getClaimsByIds(new ClaimsByIdsDto(ids));
+    return client.getClaimsByIds(new ClaimsByIdsDto(ids))
+      .stream()
+      .map(c -> {
+        val signedAudioUrl = signAudioUrl(c.getAudioURL());
+        val claimWithSigned = c.toBuilder().audioURL(signedAudioUrl).build();
```
Comment (meletios) [inline src/main/java/com/hedvig/backoffice/services/claims/ClaimsServiceImpl.java]: Shall we assign the rest of the properties in the builder also?

Comment (hedvigoscar) [reply src/main/java/com/hedvig/backoffice/services/claims/ClaimsServiceImpl.java]: I'm not actually sure the builder generates the inherited properties. I think that's why the properties were not automatically assigned by `toBuilder()`

Comment (meletios) [reply src/main/java/com/hedvig/backoffice/services/claims/ClaimsServiceImpl.java]: oh yes, it extends ClaimBackOffice 💥  👍

**Pull request merged**